src/miral/minimal_window_manager.cpp

```diff
@@ -147,7 +156,9 @@ auto miral::MinimalWindowManager::place_new_window(
     ApplicationInfo const& /*app_info*/, WindowSpecification const& requested_specification)
     -> WindowSpecification
 {
-    return requested_specification;
+    auto spec_copy = requested_specification;
+    self->try_place_new_window_and_account_for_occlusion(spec_copy);
+    return spec_copy;
```

Comment:
Nit: I would rather if `try_place_new_window_and_account_for_occlusion` would take a `WindowSpecification const&` as a parameter and return a `WindowSpecification` so that you can do:

```suggestion
    return  self>try_place_new_window_and_account_for_occlusion(requested_specification);
```